src/functional/iterator/embedded.py

```diff
@@ -567,8 +567,9 @@ def deref(self) -> Any:
         shifted_pos = self.pos.copy()
         axes = _get_axes(self.field)
 
-        if not all(axis.value in shifted_pos.keys() for axis in axes if axis is not None):
-            raise IndexError("Iterator position doesn't point to valid location for its field.")
+        if __debug__:
+            if not all(axis.value in shifted_pos.keys() for axis in axes if axis is not None):
+                raise IndexError("Iterator position doesn't point to valid location for its field.")
```

Comment:
What's this about? Performance?

Reply:
Indeed. It doesn't give that much though so I could remove it.